src/skeleplex/graph/adjacency_matrix.py

```diff
@@ -0,0 +1,179 @@
+"""Conversion of a skeletonized image into a sparse graph representation using Dask."""
+
+import functools
+import operator
+from itertools import product
+
+import dask.dataframe as dd
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy
+from dask.array.slicing import cached_cumsum
+from dask.delayed import delayed
+from scipy import sparse
+from skan.csr import _write_pixel_graph
+from skan.nputil import raveled_steps_to_neighbors
+
+
+@delayed
+def skeleton_graph_func(skelint, spacing=1):
```

Comment:
In general, I prefer variable names that are more descriptive. For example `skelint` wouldn't tell a user much about what the expected usage of that variable is. Something like `skeleton_image` would be more useful as it tells the user it is an image and it should be a skeleton. Of course, it is something that has to be balanced with the length of the name in practice. Also, to make it easier to change names, you can use the renaming functinos in your IDE (VS Code and Pycharm both support this) to automatically rename all references to the variable at the same time.

Can you please add type hints to the function signature? Type hints tell the static analysis tools what the expected types of the variables are. This is not enforced at runtime, but it helps the static analysis tools check for errors. Please do this for all other functions you added in this PR.

Some resources on type hints:
- https://fastapi.tiangolo.com/python-types/
- https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html

```suggestion
def skeleton_graph_func(skeleton_image: np.ndarray, spacing: float = 1):
```